drt_1d.py

```diff
@@ -8,121 +8,276 @@
-    return streaming_operator
-
-def construct_transition_matrix(num_voxels:int, voxel_size:float, sigma_s:np.ndarray):
-    """
-    Construct the transition matrix.
+    def compute_streaming_operator(self, start_position:float, end_position:float) -> float:
+        """
+        Computes the streaming operator between two positions.
 
-    Parameters:
-    - num_voxels (int): Number of spatial elements (voxels).
-    - voxel_size (float): Size of each voxel.
-    - sigma_s (numpy.ndarray): Vector of scattering cross-sections.
+        Parameters:
+        - start_position (float): Starting position.
+        - end_position (float): Ending position.
 
-    Returns:
-    - numpy.ndarray: Transition matrix.
-    """
-    logging.debug(f"Constructing transition matrix with {num_voxels} voxels and voxel size {voxel_size}")
-  
-    if num_voxels <= 0:
-        raise ValueError("Number of voxels must be positive and non-zero.")
-    if voxel_size <= 0:
-        raise ValueError("Voxel size must be positive and non-zero.")
+        Returns:
+        - float: Streaming operator value.
+        """
+        if start_position == end_position:
+            return 1.0 #TODO need to discuss what this value should be. Perhaps it should not even be computed and simply skipped.
+        tau = self.compute_optical_thickness(start_position, end_position)
+        streaming_operator = (1/2) * sc.exp1(tau) #TODO need to update to include voxel_size in future PR. Concerns about implementation with varying voxel_size.
+        return streaming_operator
+    
+    def construct_transition_matrix(self) -> np.ndarray:
+        """
+        Constructs the transition matrix for the domain.
+
+        Returns:
+        - np.ndarray: Transition matrix.
+        """
+        K_trans = np.zeros((self.num_voxels, self.num_voxels))
         
-    K_trans = np.zeros((num_voxels, num_voxels))
-    for i in range(num_voxels):
-        # Set diagonal values to one 
-        K_trans[i,i] = 1
-        for j in range(i):
-            K_trans[i, j] = calculate_K_trans(i, j, voxel_size, sigma_s)
-            # The transition matrix is symmetric
-            K_trans[j,i] = K_trans[i,j]
-    logging.debug(f"Constructed transition matrix: {K_trans}")
-    return K_trans
-
-def calculate_neutron_flux(K_trans:np.ndarray, sigma_s:np.ndarray, s:np.ndarray):
-    """
-    Calculate the neutron flux vector.
+        # Create a dictionary where the keys are voxel ids and values are list of positions within that voxel.
+        voxel_positions = dict()
+        for pos in self.positions:
+            voxel_id = self.get_voxel_id(pos)
+            if voxel_id not in voxel_positions:
+                voxel_positions[voxel_id] = []
+            voxel_positions[voxel_id].append(pos)
+        
+        # Now iterate through all of the combinations of voxel pairs and average the position combinations within those voxels.
+        for i in range(self.num_voxels):
```

Comment:
track the region ID as you move through the voxels to avoid having to look it up/search